views/modules.ejs

```diff
@@ -21,7 +21,7 @@
       <tr>
         <td><%= data[i].module_code%></td>
         <td>
-          <a href="#" class="btn btn-primary">
+          <a href="/courses?module_code=<%= encodeURIComponent(data[i].module_code)%>" class="btn btn-primary">
```

Comment:
Define a route helper for this. Same for `views/semesters.ejs`

Reply:
How should i name the helper?